daprdocs/content/en/js-sdk-docs/js-client/_index.md

```diff
@@ -294,14 +304,114 @@ async function start() {
   const topic = "topic-a";
 
   // Configure Subscriber for a Topic
-  await server.pubsub.subscribe(pubSubName, topic, async (data: any) => console.log(`Got Data: ${JSON.stringify(data)}`));
+  // Method 1: Direct subscribtion through the `subscribe` method
+  await server.pubsub.subscribe(pubSubName, topic, async (data: any) => console.log(`Received Data: ${JSON.stringify(data)}`));
 
+  // Method 2: Direct susbcription with options through the `subscribeWithOptions` method
+  await server.pubsub.subscribeWithOptions(pubSubName, topic, { 
+    callback: async (data: any) => console.log(`Received Data: ${JSON.stringify(data)}`) 
+  });
+
+  // Method 3: Subscription afterwards through the `susbcribeOnEvent` method
+  await server.pubsub.subscribeWithOptions('pubsub-redis', 'topic-options-1', {});
+  server.pubsub.subscribeOnEvent("pubsub-redis", "topic-options-1", "default", async (data) => { console.log(`Received Data: ${JSON.stringify(data)}`) });
+
+  // Start the server
   await server.start();
 }
 ```
 
 > For a full list of state operations visit [How-To: Publish & subscribe]({{< ref howto-publish-subscribe.md >}}).
 
+#### Subscribe to messages rule based
+
+Dapr [supports routing messages](https://docs.dapr.io/developing-applications/building-blocks/pubsub/howto-route-messages/
+) to different handlers (routes) based on rules. 
+
+> E.g., think of that you are writing an application that needs to handle messages depending on their "type" with Dapr you can then route it to `handlerType1` and `handlerType2` with a default being `handlerDefault`
+
+```javascript
+import { DaprServer } from "@dapr/dapr";
+
+const daprHost = "127.0.0.1"; // Dapr Sidecar Host
+const daprPort = "3500"; // Dapr Sidecar Port of this Example Server
+const serverHost = "127.0.0.1"; // App Host of this Example Server
+const serverPort = "50051"; // App Port of this Example Server "
+
+async function start() {
+  const server = new DaprServer(serverHost, serverPort, daprHost, daprPort);
+
+  const pubSubName = "my-pubsub-name";
+  const topic = "topic-a";
+
+  // Configure Subscriber for a Topic with rule set
+  await server.pubsub.subscribe('pubsub-redis', 'topic-1', {
+    default: "/default",
```

Comment:
What is the default value for `default` and `rules.match` parameters? We can document that.

Reply:
It is a non required parameter but I have added it as a note